lib/css_parser/rule_set.rb

```diff
@@ -113,8 +113,64 @@ def delete(property)
       end
       alias remove_declaration! delete
 
+      # Replace CSS property with multiple declarations
+      # @param [#to_s] property property name to be replaces
+      # @param [Hash<String => [String, Value]>] replacements hash with properties to replace with
+      #
+      # @example
+      #  declarations = Declarations.new('line-height' => '0.25px', 'font' => 'small-caps', 'font-size' => '12em')
+      #  declarations.replace_declaration!('font', {'line-height' => '1px', 'font-variant' => 'small-caps', 'font-size' => '24px'})
+      #  declarations
+      #  => #<CssParser::RuleSet::Declarations:0x00000000029c3018
+      #  @declarations=
+      #  {"line-height"=>#<CssParser::RuleSet::Declarations::Value:0x00000000038ac458 @important=false, @value="1px">,
+      #   "font-variant"=>#<CssParser::RuleSet::Declarations::Value:0x00000000039b3ec8 @important=false, @value="small-caps">,
+      #   "font-size"=>#<CssParser::RuleSet::Declarations::Value:0x00000000029c2c80 @important=false, @value="12em">}>
+      def replace_declaration!(property, replacements, preserve_importance: false)
+        property = normalize_property(property)
+        raise ArgumentError, "property #{property} does not exist" unless key?(property)
+
+        replacement_declarations = self.class.new(replacements)
+
+        if preserve_importance
+          importance = get_value(property).important
+          replacement_declarations.each { |_key, value| value.important = importance }
+        end
+
+        replacement_keys = declarations.keys
+        replacement_values = declarations.values
+        property_index = replacement_keys.index(property)
+
+        replacements = replacement_declarations.each.with_object({}) do |(key, value), result|
+          if !declarations.key?(key) || (value.important && !declarations[key].important)
+            result[key] = value
+            next unless declarations.key?(key)
+
+            replaced_index = replacement_keys.index(key)
+            replacement_keys.delete_at(replaced_index)
+            replacement_values.delete_at(replaced_index)
+            property_index -= 1 if replaced_index < property_index
+            next
+          end
+
+          next if declarations[key].important && !value.important
```

Comment:
looks like the the `if` above, maybe assign to a variable ?

Reply:
... or do some if + if/else flow

Reply:
it's opposite of the one above, made order the same to make it more clear.